pym/_emerge/actions.py

```diff
@@ -1644,8 +1644,30 @@ def action_info(settings, trees, myopts, myfiles):
 
 	for repo in repos:
 		last_sync = portage.grabfile(os.path.join(repo.location, "metadata", "timestamp.chk"))
+		head_commit = None
 		if last_sync:
 			append("Timestamp of repository %s: %s" % (repo.name, last_sync[0]))
+		last_sync = portage.grabfile(os.path.join(repo.location, "metadata", "timestamp.commit"))
+		if last_sync:
+			try:
+				head_commit = last_sync[0].split()[0]
+			except IndexError:
+				pass
+		elif repo.sync_type == "git":
```

Comment:
Created a method in GitSync and RsyncSync for it called retrieve_head. Tested with SVNsyncer that it'll properly raise and catch a NotImplementedError.
Please have a look at the code for _kwargs by providing a options dict to the retrieve_head function, I'm not sure if this matches with what you had in mind there.